routes/article.js

```diff
@@ -21,53 +21,37 @@ function getAmount(vote) {
 
 router.post('/vote', (req, res) => {
   const amount = getAmount(req.body.vote)
-  if (amount === 0) {
+  if (!amount) {
     res.status(500).send({
       err: 'Invalid value for vote',
       givens: req.body,
     })
     return
   }
 
-  let articleChanged = null
   Article.findById(req.body.id)
-    .then((article) => {
-      article.votes += amount
-      articleChanged = article
-      return article.save()
-    })
-    .then((data) => {
-      data.keywords.forEach((word) => {
-        Keyword.findOne({ word })
-          .then((keyword) => {
-            const votes = keyword.votes
-            const len = votes.length
-            const newVote = {
-              sum: votes[len - 1].sum + amount,
-              time: Date.now(),
-            }
-            votes.push(newVote)
-            keyword.save()
-          })
-          .catch((err) => {
-            const keyword = new Keyword({
-              word,
-              votes: [{
-                sum: amount,
-                time: Date.now(),
-              }],
-            })
-            keyword.save()
+        .then((article) => {
+          if (!article) throw new Error('Article not found')
+          article.votes += amount
+          return article.save()
+        })
+        .then((data) => {
+          Promise.all(incrementKeywords(data.keywords, amount))
+                .then((keywords) => {
+                  res.send({
+                    votes: data.votes,
+                    id: data.id,
+                    keywords,
+                  })
+                })
+                .catch(err => new Error(err.message))
```

Comment:
Eslint was complaining that err was never used, and I forgot to ask if I should ignore it or do this